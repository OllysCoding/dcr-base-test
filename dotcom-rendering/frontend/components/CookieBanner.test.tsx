import React from 'react';
import { shallow } from 'enzyme';
import CookieBanner from './CookieBanner';
import {
    getCookie as getCookie_,
    addCookie as addCookie_,
} from '../lib/cookie';

const getCookie: any = getCookie_;
const addCookie: any = addCookie_;

jest.mock('../lib/cookie', () => ({
    getCookie: jest.fn(() => null),
    addCookie: jest.fn(() => null),
}));

describe('CookieBanner', () => {
    const consentCookie = 'GU_TK';

    beforeEach(() => {
        addCookie.mockReset();
        getCookie.mockReset();
    });

    afterEach(() => {
        expect(getCookie).toHaveBeenCalledWith(consentCookie);
    });

    describe('snapshots', () => {
        it('It should render correctly if consentCookie not set', () => {
            getCookie.mockImplementation(() => false);

            const component = shallow(<CookieBanner />);

            expect(component).toMatchSnapshot();
        });

        it('It should render correctly if consentCookie set', () => {
            getCookie.mockImplementation(() => true);

            const component = shallow(<CookieBanner />);

            expect(component).toMatchSnapshot();
        });
    });

    it('It should render null if consentCookie set', () => {
        getCookie.mockImplementation(() => true);

        const component = shallow(<CookieBanner />);

        expect(component.type()).toBeNull();
        expect(component.state('show')).toBe(false);
    });

    it('It should not render null if consentCookie not set', () => {
        getCookie.mockImplementation(() => false);

        const component = shallow(<CookieBanner />);

        expect(component.type()).not.toBeNull();
        expect(component.state('show')).toBe(true);
    });

    it('It should add consentCookie on button click', () => {
        getCookie.mockImplementation(() => false);

        const component = shallow(<CookieBanner />);
        const buttonElem = component.find('button').first();

        expect(component.type()).not.toBeNull();
        expect(component.state('show')).toBe(true);

        buttonElem.simulate('click');

        expect(component.state('show')).toBe(false);
        expect(addCookie).toHaveBeenCalled();
    });
});
