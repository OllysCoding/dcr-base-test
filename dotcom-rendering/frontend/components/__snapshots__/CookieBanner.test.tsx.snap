// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBanner snapshots It should not render if consentCookie set 1`] = `""`;

exports[`CookieBanner snapshots It should render if consentCookie not set 1`] = `
.emotion-5 {
  position: fixed;
  bottom: 0;
  background-color: #005689;
  padding: 10px 0 24px;
  width: 100%;
}

.emotion-5 p {
  font-family: GuardianTextEgyptian,Georgia,serif;
  font-size: 17px;
  line-height: 24px;
  margin-top: 0;
  margin-bottom: 8px;
  color: #ffffff;
}

.emotion-5 a {
  color: #ffffff;
  border-bottom: 1px solid rgba(255,255,255,0.5);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 a:hover {
  border-color: #ffffff;
}

.emotion-4 {
  padding: 0 10px;
}

.emotion-0 {
  font-size: 34px;
  line-height: 38px;
  font-weight: bold;
  padding-bottom: 12px;
  color: #ffe500;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-family: GuardianTextSans,Helvetica Neue,Helvetica,Arial,Lucida Grande,sans-serif;
  margin-left: 12px;
}

.emotion-1 {
  font-size: 16px;
  line-height: 24px;
  font-family: GuardianTextSans,Helvetica Neue,Helvetica,Arial,Lucida Grande,sans-serif;
  border-radius: 1000px;
  height: 42px;
  background: #ffe500;
  color: #121212;
  padding: 0 25px 0 46px;
  display: block;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  position: relative;
  border: 0;
  white-space: nowrap;
  display: inline-block;
}

.emotion-1 svg {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  margin: auto;
  width: 42px;
  height: 42px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

.emotion-3 {
  margin-top: 24px;
}

<div
  className="emotion-5"
>
  <Container
    className="emotion-4"
  >
    <h1
      className="emotion-0"
    >
      Your privacy
    </h1>
    <p>
      We use cookies to improve your experience on our site and to show you relevant advertising.
    </p>
    <p>
      To find out more, read our updated
       
      <a
        href="/help/privacy-policy"
      >
        privacy policy
      </a>
       and
       
      <a
        href="/info/cookies"
      >
        cookie policy
      </a>
      .
    </p>
    <div
      className="emotion-3"
    >
      <button
        className="emotion-1"
        onClick={[Function]}
      >
        <SVG />
        OK
      </button>
      <a
        className="emotion-2"
        href="https://profile.theguardian.com/privacy-settings"
      >
        More information
      </a>
    </div>
  </Container>
</div>
`;
